net/pfSense-pkg-pfBlockerNG-devel/files/usr/local/pkg/pfblockerng/pfblockerng.inc

```diff
@@ -319,15 +319,21 @@ function pfb_filter_service() {
 		/bin/ln "\${php_path}/php" "\${php_path}/php_pfb"
 	fi
 
-	# Start pfBlockerNG Firewall filter Daemon
-	if [ -e '/var/log/filter.log' ]; then
-		/usr/local/sbin/clog_pfb -f /var/log/filter.log | /usr/local/bin/php_pfb -f /usr/local/pkg/pfblockerng/pfblockerng.inc filterlog &
-	fi
+        # Start pfBlockerNG Firewall filter Daemon
+        file="/usr/local/sbin/clog"
+        if [ -e '/var/log/filter.log']; then
+                if [ -f "$file" ]
+                then
+                /usr/local/sbin/clog_pfb -f /var/log/filter.log | /usr/local/bin/php_pfb -f /usr/local/pkg/pfblockerng/pfblockerng.inc filterlog &
+                fi
+        else
+                /usr/bin/tail_pfb -f /var/log/filter.log | /usr/local/bin/php_pfb -f /usr/local/pkg/pfblockerng/pfblockerng.inc filterlog &
```

Comment:
You cannot tail if /var/log/filter.log doesn't exist. This `else` conditional belongs to inner `if`. Also, you can simplify declaring which program should be used on a variable and deduplicate code. You will end up with something like:

```shell
# Start pfBlockerNG Firewall filter Daemon
if [ -e '/var/log/filter.log' ]; then
	# Decide about using clog or tail
	if [ -f /usr/local/sbin/clog ]; then
		tail_binary="/usr/local/sbin/clog_pfb"
	else
		tail_binary="/usr/bin/tail_pfb"
	fi

	${tail_binary} -f /var/log/filter.log \
	    | /usr/local/bin/php_pfb \
	    -f /usr/local/pkg/pfblockerng/pfblockerng.inc \
	    filterlog &
fi
```